hardware/raspberrypi/booteeprom.md

```diff
@@ -2,82 +2,118 @@
-
-## USB boot
-
-USB boot support is currently being beta-tested. See the "USB mass storage boot" section of the [Bootloader Configuration Page](./bcm2711_bootloader_config.md) for further details.
-
-## Is the bootloader working correctly?
-
-To check that the bootloader is working correctly, turn off the power, unplug everything from the Raspberry Pi 4, including the SD card, and then turn the power back on. If the green LED blinks with a repeating pattern then the bootloader is running correctly, and indicating that `start*.elf` has not been found. Any other actions imply that the bootloader is not working correctly and should be reinstalled using `recovery.bin`.
-
-## Recovery image
+## Updating the bootloader
 
-If the Raspberry Pi is not booting it's possible that the bootloader EEPROM is corrupted. This can easily be reprogrammed using the Raspberry Pi Imager tool which is available via the [raspberrypi.org downloads page](https://www.raspberrypi.org/downloads/).
+### Raspberry Pi Imager
+The easiest way to to update the bootloader to the latest version with default settings is to use the [Raspberry Pi Imager](https://www.raspberrypi.org/downloads/) to install a boot recovery image onto a spare SD card.
 
-Using the recovery image will erase any custom configuration options, resetting the bootloader back to factory defaults.
+Select "Choose OS -> Misc utility images -> Raspberry Pi 4 EEPROM boot recovery"
 
-## Updating the bootloader
+Raspberry Pi OS also keeps the EEPROM up-to-date: new EEPROM images are applied when the Raspberry Pi 4 next boots.
 
+### Updating from Raspberry Pi OS
 Bootloader updates are instigated during a normal `apt update`, `apt full-upgrade` cycle, this means you will get new features and bug fixes during your normal updates. 
 
-Bootloader updates are performed by the `rpi-eeprom` package, which installs a service that runs at boot-time to check for critical updates. 
+Bootloader updates are performed by `rpi-eeprom-update` service provided by the `rpi-eeprom` package. This service runs at boot and updates the bootloader at the next reboot if a new production release is available. The service automatically migrates the current boot settings to the new bootloader release.
 
 To update your system, including the bootloader:
-
 ```
 sudo apt update
 sudo apt full-upgrade
 sudo reboot
 ```
 
-If you wish to control when the updates are applied you can disable the systemd service from running automatically and run `rpi-eeprom-update` manually, as shown in the 'Manually checking if an update is available' section below.
+## Manually checking if an update is available
+Running the `rpi-eeprom-update` command with no parameters indicates whether an update is required. An update is required if the timestamp of the most recent file in the firmware directory (normally `/lib/firmware/raspberrypi/bootloader/critical`) is newer than that reported by the current bootloader.
```

Comment:
Might be worth clarifying that this "timestamp" is embedded inside the file itself, and isn't the last-modified timestamp of the file? Alternatively you could just replace "timestamp" with "version"?

Perhaps "in the firmware directory" should be "in the relevant firmware directory" ?

Reply:
replaced with version